metaseq/criterions/cross_entropy.py

```diff
@@ -209,13 +210,19 @@ def serialize_tensor_to_numpy(t):
                         batch_size = logging_output["targets"].shape[0]
                         for i in range(batch_size):
 
+                            num_pad = sum(int(x==1) for x in serialize_tensor(logging_output["targets"][i]))
+                            length_final = len(serialize_tensor(logging_output["log_probs"][i])),
+                            
+
                             with open(f"{data_pruning_metrics_savedir}/ssl_embeddings/{counter}_emb_{hash_to_add}.npy", "wb") as embedding_out_f:
                                 serialized_embedding = serialize_tensor_to_numpy(logging_output["final_embedding"][i])
                                 np.save(embedding_out_f, serialized_embedding)
 
                                 log_line = {
                                     "name":  f"{data_pruning_metrics_savedir}/ssl_embeddings/{counter}_emb_{hash_to_add}.npy",
                                     "path_info": logging_output["path_infos"][i],
+                                    "length": length_final,
+                                    "num_pad": num_pad
```

Comment:
adding the num pad / length computation because otherwise it takes kind of long to produce el2n scores (because we have to load in the ppl csv file and search through it)